### ISSUE TITLE
try_inverse has a trait bound that is too restrictive (i.e. Real)

### ISSUE THREAD

**Issue by vjackson725** [sebcrozet/nalgebra#303]
# Summary
At the moment, the functions for LU decomposition, and try_inverse have a trait bound of Real. I can't think of any technical reasons (although there might exist some) not to just use Field.

This also might be applicable to the decomposition algorithms, although I'm not familiar with their details.

# Motivation
If the trait bound is loosened, it opens up the ability to work with matrices on different Fields. For example, I want to work with rationals, so I don't have to worry about floating point arithmetic errors.

# Drawbacks
* This might conflict with any optimizations you may have for floating point numbers, if you have any. However, this should be able to be solved with specialization.
* There might be some technical reason I'm not aware of why this won't work.

**Comment by aepsil0n** [comment]
Just to be clear: you are talking about the `try_inverse` method defined on `SquareMatrix`, right? (There are a couple more with varying trait bounds.

**Comment by vjackson725** [comment]
Yes. `try_inverse` as on `SquareMatrix` has a bound of `Real`. This is also the case with the decompositions.

Additionally, the `try_inverse` in the base level package only needs `AlgaSquareMatrix`, but that trait doesn't seem to be implemented by anything in nalgebra.
